app/models/collection.rb

```diff
@@ -86,6 +92,29 @@ def submitters
     User.with_role(:submitter, self)
   end
 
+  # Permit a User to receive notification messages for members of this Collection
+  # @param user [User]
+  def enable_messages_for(user:)
+    raise(ArgumentError, "User #{user.uid} is not an administrator for this collection #{title}") unless user.super_admin? || user.can_admin?(self)
+    collection_messaging_options << CollectionOption.new(option_type: CollectionOption::EMAIL_MESSAGES, collection: self, user: user)
+  end
+
+  # Disable a User from receiving notification messages for members of this Collection
+  # @param user [User]
+  def disable_messages_for(user:)
+    raise(ArgumentError, "User #{user.uid} is not an administrator for this collection #{title}") unless user.super_admin? || user.can_admin?(self)
+    users_with_messaging.destroy(user)
+  end
+
+  # Returns true if a given user has notification e-mails enabled for this collection
+  # @param user [User]
+  # @return [Boolean]
+  def messages_enabled_for?(user:)
+    found = users_with_messaging.find_by(id: user.id)
+
+    !found.nil?
```

Comment:
I find positive checks easier to read in the long run.
```suggestion
    found.present?
```